src/shared/creds-util.c

```diff
@@ -668,6 +671,12 @@ int encrypt_credential_and_warn(
                         log_debug("Running in container, not attempting to use TPM2.");
 
                 try_tpm2 = r <= 0;
+
+                support = tpm2_support();
+                if (support != TPM2_SUPPORT_FULL) {
+                        log_debug("System lacks TPM2 support, not attempting to use TPM2.");
+                        try_tpm2 = false;
+                }
```

Comment:
tpm2_support() internally already does a container check, hence you can drop that part now

Reply:
Yes. I kept that because of the specific error message. Now I dropped the code and updated the comment and the debug mesage